packages/packages/core/editor-components/src/components/components-tab/search-provider.tsx

```diff
@@ -0,0 +1,38 @@
+import * as React from 'react';
+import { createContext, useContext } from 'react';
+import { useSearchState, type UseSearchStateResult } from '@elementor/utils';
+
+type SearchContextType = Pick< UseSearchStateResult, 'handleChange' | 'inputValue' > & {
+	searchValue: UseSearchStateResult[ 'debouncedValue' ];
+	onClearSearch: () => void;
+};
+
+const SearchContext = createContext< SearchContextType | undefined >( undefined );
+
+export const SearchProvider = ( {
+	children,
+	localStorageKey,
+}: {
+	children: React.ReactNode;
+	localStorageKey: string;
+} ) => {
+	const { debouncedValue, handleChange, inputValue } = useSearchState( { localStorageKey } );
+
+	const onClearSearch = () => {
```

Comment:
```suggestion
	const clearSearch = () => {
```
since it the actual action and not a callback